workos/resources/list.py

```diff
@@ -0,0 +1,74 @@
+from workos.utils.pagination_order import Order
+from workos.utils.auto_pagination import auto_paginate, timestamp_compare
+
+
+class WorkOSListResource(object):
+    """Representation of a WorkOS List Resource as returned through the API.
+
+    Attributes:
+        OBJECT_FIELDS (list): List of fields a WorkOSListResource is comprised of.
+    """
+
+    OBJECT_FIELDS = [
+        "data",
+        "list_metadata",
+    ]
+
+    @classmethod
+    def construct_from_response(cls, response):
+        """Returns an instance of WorkOSListResource.
+
+        Args:
+            response (dict): Resource data from a WorkOS API response
+
+        Returns:
+            WorkOSListResource: Instance of a WorkOSListResource with OBJECT_FIELDS fields set
+        """
+        obj = cls()
+        for field in cls.OBJECT_FIELDS:
+            setattr(obj, field, response.get(field))
+
+        return obj
+
+    def to_dict(self):
+        """Returns a dict representation of the WorkOSListResource.
+
+        Returns:
+            dict: A dict representation of the WorkOSListResource
+        """
+        obj_dict = {}
+        for field in self.OBJECT_FIELDS:
+            obj_dict[field] = getattr(self, field, None)
+
+        return obj_dict
+
+    def auto_paging_iter(self):
```

Comment:
Why is there specific logic for each list type? I think the aim here should be a generic list type that could be applied to any object. Can we make this generic - so really just pass through any list options and return all the objects in a given list?

The idea is that when we add new objects, we can just apply a generic list and auto-pagination strategy. So the logic here should not rely on the type of object in the list.

Reply:
Some more context here - the auto pagination method of the list object should just flow through any parameters provided and call the list method for that object, iterating through all the pages, and returning all the resources that apply to the list query.

The only parameters that the auto pagination method needs to adjust are the pagination specific ones that allow it to iterate through the pages of resources returned. But the other parameters should be applied in a generic way by the method.

In this way, the method can be generic, because the pagination parameters are the same across list resources. The pagination should only deal with generic parameters, and just pass through any resource-specific parameters.

Reply:
Good call. I think this was more of a holdover from how I had initially constructed the method. I refactored it to be more generic at the WorkOSListResource level.